src/FluentValidation.AspNetCore/RuleSetForClientSideMessagesAttribute.cs

```diff
@@ -1,59 +1,46 @@
 namespace FluentValidation.AspNetCore {
 	using System;
-  using FluentValidation.Internal;
-  using Microsoft.AspNetCore.Http;
+	using Microsoft.AspNetCore.Http;
 	using Microsoft.AspNetCore.Mvc.Filters;
 
 	/// <summary>
 	/// Specifies which ruleset should be used when deciding which validators should be used to generate client-side messages.
 	/// </summary>
 	public class RuleSetForClientSideMessagesAttribute : ActionFilterAttribute {
-		private const string _key = "_FV_ClientSideRuleSet";
-		private readonly string[] _ruleSets;
 
-		public RuleSetForClientSideMessagesAttribute(string ruleSet) {
-			_ruleSets = new[] { ruleSet };
-		}
+		private readonly string[] _ruleSets;
 
-		public RuleSetForClientSideMessagesAttribute(params string[] ruleSets) {
-			this._ruleSets = ruleSets;
-		}
+		public RuleSetForClientSideMessagesAttribute(string ruleSet) => _ruleSets = new[] { ruleSet };
 
-		public override void OnActionExecuting(ActionExecutingContext filterContext) {
-			var contextAccessor = filterContext.HttpContext.RequestServices.GetService(typeof(IHttpContextAccessor));
+		public RuleSetForClientSideMessagesAttribute(params string[] ruleSets) => _ruleSets = ruleSets;
 
-			if(contextAccessor == null) {
-				throw new InvalidOperationException("Cannot use the RuleSetForClientSideMessagesAttribute unless the IHttpContextAccessor is registered with the service provider. Make sure the provider is registered by calling services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>(); in your Startup class's ConfigureServices method");
-			}
+		public override void OnActionExecuting(ActionExecutingContext filterContext) => SetRulesetOnExecuting(filterContext);

```

Comment:
Do we need `OnActionExecuting` at all anymore, now that we have `OnResultExecuting` overridden? I think having both will mean `SetRulesetOnExecuting` will end up running twice for MVC actions.

Reply:
You are correct. Not needed so I have pushed a commit which removes `OnActionExecuting`